libpod/container_internal_linux.go

```diff
@@ -1241,6 +1242,26 @@ func (c *Container) makeBindMounts() error {
 		c.state.BindMounts["/etc/hostname"] = hostnamePath
 	}
 
+	// Make /etc/localtime
+	if _, ok := c.state.BindMounts["/etc/localtime"]; !ok {
```

Comment:
Only do this is timezone is not set to "" - we should not modify behavior for containers not explicitly passing the flag.